src/tokenizer.rs

```diff
@@ -74,20 +75,30 @@ where
             next_char = self.chars.next()?
         }
 
-        match next_char {
-            '\0' => None,
-            '(' => Some(Token::ParenOpen),
-            ')' => Some(Token::ParenClose),
-            ',' => Some(Token::Comma),
+        let token = match next_char {
+            '\0' => return None,
+            '(' => Token::ParenOpen,
+            ')' => Token::ParenClose,
+            ',' => Token::Comma,
             c if is_numberlike(c) => {
                 let number = self.read_until_whitespace(if c == '+' { None } else { Some(c) });
                 match number.parse::<T>() {
-                    Ok(parsed_num) => Some(Token::Number(parsed_num)),
-                    Err(_) => None,
+                    Ok(parsed_num) => Token::Number(parsed_num),
+                    Err(_) => {
+                        log::warn!(
+                            "Failed to parse input: '{}' as {}",
+                            &number,
+                            type_name::<T>()
+                        );
+                        return Some(Err(
+                            "Unable to parse input number as the desired output type",
```

Comment:
This is probably not a number, right?

Actually, we might as well include it in the error instead of pulling `log` along.

Reply:
It is a `numberlike` as per line 83. In particular to this PR is the case when trying to parse "1.23" as an integer, which will fail with this message.

The reason I logged here rather than building a `String` with the error details to the error is that errors throughout the WKT library are `&'static str`. I could do something different here, but it'd affect multiple places since this error is propagated.